docs/linting/troubleshooting/Performance.md

```diff
@@ -0,0 +1,150 @@
+---
+id: performance-troubleshooting
+title: Performance Troubleshooting
+---
+
+As mentioned in the [type-aware linting doc](../Typed_Linting.md), if you're using type-aware linting, your lint times should be roughly the same as your build times.
+
+If you're experiencing times much slower than that, then there are a few common culprits.
+
+## Wide includes in your `tsconfig`
+
+When using type-aware linting, you provide us with one or more tsconfigs.
+We then will pre-parse all files so that full and complete type information is available.
+
+If you provide very wide globs in your `include` (such as `**/*`), it can cause many more files than you expect to be included in this pre-parse.
+Additionally, if you provide no `include` in your tsconfig, then it is the same as providing the widest glob.
+
+Wide globs can cause TypeScript to parse things like build artifacts, which can heavily impact performance.
+Always ensure you provide globs targeted at the folders you are specifically wanting to lint.
+
+## Wide includes in your ESLint options
+
+Specifying `tsconfig.json` paths in your ESLint commands is also likely to cause much more disk IO than expected.
+Instead of globs that use `**` to recursively check all folders, prefer paths that use a single `*` at a time.
+
+```js title=".eslintrc.js"
+module.exports = {
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+  ],
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    tsconfigRootDir: __dirname,
+    // Remove this line
+    project: ['./**/tsconfig.json'],
+    // Add this line
+    project: ['./packages/*/tsconfig.json'],
+  },
+  plugins: ['@typescript-eslint'],
+  root: true,
+};
+```
+
+See [Glob pattern in parser's option "project" slows down linting](https://github.com/typescript-eslint/typescript-eslint/issues/2611) for more details.
+
+## The `indent` / `@typescript-eslint/indent` rules
+
+This rule helps ensure your codebase follows a consistent indentation pattern.
+However this involves a _lot_ of computations across every single token in a file.
+Across a large codebase, these can add up, and severely impact performance.
+
+We recommend not using this rule, and instead using a tool like [`prettier`](https://www.npmjs.com/package/prettier) to enforce a standardized formatting.
+
+See our [documentation on formatting](./Formatting.md) for more information.
```

Comment:
The file is `FORMATTING.md` but the link here is `Formatting.md`. I'd prefer renaming the file to `Formatting.md`.

Reply:
ugh damn case insensitive filesystems.
i did rename it but git didn't pick it up.
i'll have to do an explicit `git mv`